vispy/visuals/volume.py

```diff
@@ -754,10 +747,14 @@ def clipping_planes(self):
         a set of planes used to clip the volume. Each plane is defined by a position and
         a normal vector (magnitude is irrelevant). Shape: (n_planes, 2, 3)
         """
-        return self._clipping_planes
+        if self._clipping_planes is None:
+            return self._clipping_planes
+        return self._clipping_planes[:, :, ::-1]
```

Comment:
So why do this but then use the private attribute elsewhere?

Reply:
because when we get the public attribute we also reverse the last dimension of the private attribute - I agree it's weird, but this way you get to set/get the clipping planes defined according to the volume dimensions
i.e.
vol zyx
planes zyx/zyx

and everything renders as expected internally

Reply:
cause everything is zyx in glsl but xyz in vispy?

Reply:
not sure exactly where things get switched around I think if we say the numpy array is zyx that gets passed directly to the shader and anything in glsl needs to then be ordered xyz

see https://github.com/vispy/vispy/blob/5efc22184a60c7cf3f1560af74eb10dc675ad638/vispy/visuals/volume.py#L636-L637